Objects/object.c

```diff
@@ -2908,13 +2908,11 @@ Py_ReprLeave(PyObject *obj)
 void
 _PyTrash_thread_deposit_object(PyThreadState *tstate, PyObject *op)
 {
-    _PyObject_ASSERT(op, _PyObject_IS_GC(op));
-    _PyObject_ASSERT(op, !_PyObject_GC_IS_TRACKED(op));
     _PyObject_ASSERT(op, Py_REFCNT(op) == 0);
 #ifdef Py_GIL_DISABLED
     op->ob_tid = (uintptr_t)tstate->delete_later;
 #else
-    _PyGCHead_SET_PREV(_Py_AS_GC(op), (PyGC_Head*)tstate->delete_later);
+    *((PyObject**)op) = tstate->delete_later;
```

Comment:
Line 2915 is a little obscure.  I think a comment explaining that `ob_refcnt` is being used to create the queue would be helpful.